### PR TITLE
Max: updated implementation of save_scene + small QOL improvements to host

### PR DESCRIPTION
**Pull request by Sponge96**
## Changelog Description
Ported version of: https://github.com/ynput/OpenPype/pull/6186

## Testing notes:
1. Start 3dsmax
2. Check that on the right hand side it has defaulted to the 'create' view/menu/mode
3. Open the 'Publish/Create' AYON Menu
4. Create new 'workfile' publish
5. Publish
6. Check the log of save_scene is NOT saving if no changes to scene were made; or is saving when changes have been made (it should display INFO or DEBUG information in either case).

### CODE REVIEW

**Comment by mkolar** [comment]
Thanks for the port @Sponge96 as soon as we test this, we'll merge both. Also here's a medal 🏅 for the first community contribution to the brand new ayon-core repo :D

File: client/ayon_core/hosts/max/plugins/publish/save_scene.py
Comment (moonyuet) [inline client/ayon_core/hosts/max/plugins/publish/save_scene.py:23]:
```suggestion
            self.log.debug("No unsaved changes, skipping file save..")
```

File: client/ayon_core/hosts/max/plugins/publish/save_scene.py
Comment (moonyuet) [inline client/ayon_core/hosts/max/plugins/publish/save_scene.py:2]:
```suggestion
from ayon_core.pipeline import registered_host
```

**Comment by moonyuet** [comment]
@Sponge96 I have commented some of the codes and I think it's good to merge after resolving them as well as the conflicts.

**Comment by Sponge96** [comment]
> @Sponge96 I have commented some of the codes and it's good to merge after resolving them as well as the conflicts.

Thanks Kayla. Haven't had a chance to setup ayon-core properly for development yet so apologies for missing those bits. Everything should be resolved now, might be good to run black over `pipeline.py` as well but didn't want to add to the changes.

**Review by moonyuet — APPROVED** [review]
Thank you! It's good for merging

**Pull request merged**
